### PR TITLE
Todo List Structure

### PR DESCRIPTION
**Pull request by Bettenoch**
Made the following changes♻:

    🔹Setup project with Webpack and set rules for loaders.
    🔸create index.js file, index.css and index.html files.
    🔹Create todo list structure on index.html
    🔸Create modules for storing todo list array with description, completed and index keys.
    🔹Create function to load todos from local storage.
    🔸Styled page elements accordingly.

Thank You for reviewing my repository!!🙏🙏☺️

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+});
+
+taskField.addEventListener('keypress', (e) => {
+  if (e.target.className === 'todo-task' && e.key === 'Enter') {
+    if (e.target.textContent) {
+      e.preventDefault();
+      newTasks.updateTask(e.target.textContent, e.target.parentElement.id);
+      store();
+    } else {
+      e.preventDefault();
```
Comment (V-Blaze) [inline src/index.js:42]: - Your project looks good and functions well! However, I noticed that you completed the entire project in one milestone. While it's impressive, it's generally not the best approach. So please consider working on the specific milestone which will allow you to work on specific tasks, request review and make changes upon review, and update them over time. :+1: :waffle: 

For this specific mission, a few adjustments are still needed:

- [ ] As per the project requirements, in the `index.js` file set an array of some simple to-do tasks (array of objects). Each task object should contain three keys: description [string], completed [bool], and index [number]. :+1: 

 - [ ] In the index.js file, write a function to iterate over the tasks array and dynamically populate an HTML list item element for each task. :waffle: :+1: 

 - [ ] Upon page load, render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the index values for each task.  :+1: :100: 

By making these adjustments and following the project requirements, your project will be even more polished and organized. Keep up the great work, and I'm excited to see how your project evolves through milestones! :rocket:

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable linebreak-style */
```
Comment (V-Blaze) [inline src/index.js:1]: - [ ] As required by Microverse, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter disable rule and fix the issue. :+1: :+1: :+1:

File: src/modules/DisplayTask.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable new-cap */
```
Comment (V-Blaze) [inline src/modules/DisplayTask.js:2]: - [ ] As I have suggested above, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

File: src/modules/AddTask.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable linebreak-style */
```
Comment (V-Blaze) [inline src/modules/AddTask.js:1]: - [ ] As i have suggested above, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

File: src/modules/TaskList.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable linebreak-style */
```
Comment (V-Blaze) [inline src/modules/TaskList.js:1]: - [ ] As i have suggested above, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

File: src/modules/LocalStorage.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable linebreak-style */
```
Comment (V-Blaze) [inline src/modules/LocalStorage.js:1]: - [ ] As i have suggested above, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Bettenoch , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Webpack setup correctly ✔
-  Correct Github flow👋 
-  Readme looks professional✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/LocalStorage.js (hunk 6 above)
Comment (Bettenoch) [reply src/modules/LocalStorage.js:1]: Hello, @V-Blaze, thank you for reviewing my code. I have an issue with the eslint error "Expected line breaks to be LF but found CRLF". I have tried fixing even in my previous projects  but it is still recurring on all JS FILES. Could you suggest any changes on that.

Comment (Bettenoch) [reply src/modules/LocalStorage.js]: Thank you I finally managed to solve the eslint error.

Comment (Abenezer-Tilahun) [reply src/modules/LocalStorage.js]: Fixed.

File: src/modules/TaskList.js (hunk 5 above)
Comment (Abenezer-Tilahun) [reply src/modules/TaskList.js]: Fixed.

File: src/modules/AddTask.js (hunk 4 above)
Comment (Abenezer-Tilahun) [reply src/modules/AddTask.js]: Fixed.

File: src/modules/DisplayTask.js (hunk 3 above)
Comment (Abenezer-Tilahun) [reply src/modules/DisplayTask.js]: Fixed.

File: src/index.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply src/index.js]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Bettenoch 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**